docs/index.md

```diff
@@ -145,9 +140,57 @@ illustrates how to do this:
     	$ unset DOCKER_TLS_VERIFY
       ```
 
+## Discovery (optional)
+
+Instead of providing a comma-separated list of machines, Swarm can be configured to
+dynamically discover machines.
+
+This allows to add and remove machines from the cluster without having to reconfigure and restart
+the Swarm manager.
+
+For this to work, each machine will run a Swarm agent. The agent registers the referenced
+Docker daemon, monitors it, and updates the discovery backend with the machine's status.
```

Comment:
Not sure, but recent docs/communications address the daemon as "Docker Engine". Does that apply here?